File: src/main/scala/com/bjsxt/sparkscalacode/SparkBroadCast.scala
package com.bjsxt.sparkscalacode

import org.apache.spark.broadcast.Broadcast
import org.apache.spark.{SparkConf, SparkContext}

object SparkBroadCast {
  def main(args: Array[String]): Unit = {
    val sc = new SparkContext(new SparkConf().setMaster("local").setAppName("test"))

    val blackName = List[String]("zs","list")

    val bc: Broadcast[List[String]] = sc.broadcast(blackName)
    val bc2: Broadcast[List[String]] = sc.broadcast(List("zs","list"))

    val nameRDD = sc.parallelize(Array[String]("zs","lisi","wangwu"))

    nameRDD.filter(s=>{
      val value: List[String] = bc.value
      !value.contains(s)
    }).foreach(println)
  }
}

File: src/main/java/com/bjsxt/sparkjavacode/SparkBroadCast.java
package com.bjsxt.sparkjavacode;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.broadcast.Broadcast;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SparkBroadCast {
    public static void main(String[] args) {
        JavaSparkContext sc = new JavaSparkContext(new SparkConf().setMaster("local").setAppName("test"));

        List<String> list = new ArrayList<>();
        list.add("zs");
        list.add("lisi");
        list.add("wangwu");
        Broadcast<List<String>> broadcast = sc.broadcast(list);

        sc.parallelize(Arrays.asList("zs", "lisi", "wangwu", "xiaoliu")).
                filter(s -> !broadcast.value().contains(s)).
                foreach(s-> System.out.println(s));

    }
}
